airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeSqlOperations.java

```diff
@@ -20,19 +22,21 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeSqlOperations.class);
 
   @Override
-  public void createTableIfNotExists(final JdbcDatabase database, final String schemaName, final String tableName) throws SQLException {
-    final String createTableQuery = String.format(
+  public String createTableQuery(final JdbcDatabase database, final String schemaName, final String tableName) {
+    return String.format(
         "CREATE TABLE IF NOT EXISTS %s.%s ( \n"
             + "%s VARCHAR PRIMARY KEY,\n"
             + "%s VARIANT,\n"
             + "%s TIMESTAMP WITH TIME ZONE DEFAULT current_timestamp()\n"
             + ") data_retention_time_in_days = 0;",
         schemaName, tableName, JavaBaseConstants.COLUMN_NAME_AB_ID, JavaBaseConstants.COLUMN_NAME_DATA, JavaBaseConstants.COLUMN_NAME_EMITTED_AT);
-    database.execute(createTableQuery);
   }
 
+  @Override
   public boolean isSchemaExists(JdbcDatabase database, String outputSchema) throws Exception {
-    return database.query(SHOW_SCHEMAS).map(schemas -> schemas.get(NAME).asText()).anyMatch(outputSchema::equalsIgnoreCase);
+    try (final Stream<JsonNode> results = database.query(SHOW_SCHEMAS)) {
```

Comment:
Snowflake no longer opens a new connection for every query, so now there is a maximum number of connections (`5`) available in the connection pool. The `query` method returns a stream which will only close the connection when the stream is explicitly closed. If we don't close the stream here, the connector will hang after a few query operations because the connection pool would run out of available connections.